src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -162,8 +186,12 @@ public static ParseResult Parse(string expressionText, Features features, Parser
         /// <returns></returns>
         public CheckResult Check(string expressionText, RecordType parameterType, ParserOptions options = null)
```

Comment:
>expressionText [](http://example.com/codeflow?start=40&length=14)

Public API, we should test expressionText is not null, below max length

Reply:
This is just one path through the parser. We check it at a lower level in the  stack (in CheckWorker) to gaurantee we catch all paths.